scripts/validate_systs.py

```diff
@@ -0,0 +1,382 @@
+    )
+    delta_up = np.asarray([h * nom - nom for h in hi])
+    delta_dn = np.asarray([l * nom - nom for l in lo])
+    norm_deltas = handle_deltas(delta_up, delta_dn)
+    norm_rel_size = np.absolute(norm_deltas) / (1.0 * nom)
+
+    sys_names = sys_names_histo + sys_names_norm
+
+    for iBin in range(n_bins):
+        bin_number = iBin + 1
+        rel_size = np.concatenate((histo_rel_size[:, iBin], norm_rel_size[:, iBin]))
+        sys_names_bin = [
+            sys_names[idx]
+            for idx, size in enumerate(rel_size)
+            if size > 0.5 and np.isfinite(size)
+        ]  # Select for relative systematics above 0.5 (i.e. >50% variation from nominal)
+        rel_size = rel_size[(rel_size > 0.5) & (np.isfinite(rel_size))]
+
+        # Only make a plot if there are any relative systematics above 0.5
+        if len(sys_names_bin) > 0:
+            x = np.arange(len(sys_names_bin))  # the label locations
+            width = 0.5  # the width of the bars
+            fig = plt.figure(figsize=(12, 6))
+            ax = fig.add_subplot(1, 1, 1)
+            ax.set_ylabel("Relative Syst Size", fontsize=16)
+            ax.set_title(
+                f"{signal_name}: {channel_name} (Bin {bin_number}), Norm & Histo",
+                fontsize=20,
+            )
+            ax.bar(x - width / 2, rel_size, width)
+            ax.set_xticks(x)
+            ax.set_xticklabels(sys_names_bin, rotation=45, ha="right")
+            plt.tight_layout()
+            plt.savefig(
+                f"Plots/rel_systs_{signal_name}_{channel_name}_bin{bin_number}.png"
+            )
+            plt.close()
+
+
+@click.command()
+@click.option(
+    "--signal_template",
+    help="Signal name template, with signal masses as variables. Signal masses must be denoted by {a}, {b}, {c}, ... (eg. signal_{a}_{b}_more_info)",
+)
+@click.option(
+    "--x_var",
+    help="Which variable from the signal name template to plot on the x axis (defaults to 'a')",
+    default="a",
+    required=False,
+)
+@click.option(
+    "--y_var",
+    help="Which variable from the signal name template to plot on the y axis (defaults to 'b')",
+    default="b",
+    required=False,
+)
+@click.option(
+    "--v_max",
+    help="Maximum amplitude of the colourbar for plotting interpolated relative syst sizes",
+    default=10,
+    required=False,
+)
+@click.option(
+    "--x_label",
+    help="x label for interpolated plot of relative systematics",
+    default=None,
+    required=False,
+)
+@click.option(
+    "--y_label",
+    help="y label for interpolated plot of relative systematics",
+    default=None,
+    required=False,
+)
+def outlier_plot(signal_template, v_max, x_var, y_var, x_label, y_label):
+
+    patches = [json.load(open(x)) for x in glob.glob("patch*.json")]
+    # grouped_patches = {
+    #    x[0]["value"]["name"]: {p["path"]: p for p in x if p["op"] == "add"}
+    #    for x in patches
+    #    if "value" in x[0]
+    # }
```

Comment:
```suggestion
```
Unless this code is needed for something I'd delete all commented out code chunks.